go/vt/vtgate/semantics/typer.go

```diff
@@ -47,7 +47,7 @@ func (t *typer) up(cursor *sqlparser.Cursor) error {
 		t.m[node] = evalengine.NewType(node.SQLType(), collations.CollationForType(node.SQLType(), t.collationEnv.DefaultConnectionCharset()))
 	case *sqlparser.Argument:
 		if node.Type >= 0 {
-			t.m[node] = evalengine.NewType(node.Type, collations.CollationForType(node.Type, t.collationEnv.DefaultConnectionCharset()))
+			t.m[node] = evalengine.NewTypeEx(node.Type, collations.CollationForType(node.Type, t.collationEnv.DefaultConnectionCharset()), true, node.Size, node.Scale)
```

Comment:
We have to watch if this one or #15783 merges first, as the signature of this function changes to allow for passing in enum / set values.

Reply:
https://github.com/vitessio/vitess/pull/15783 is merged now, so please make sure to merge main / rebase this PR.